torch/nn/modules/loss.py

```diff
@@ -108,40 +108,61 @@ class NLLLoss(_WeightedLoss):
     The `input` given through a forward call is expected to contain
     log-probabilities of each class. `input` has to be a Tensor of size either
     :math:`(minibatch, C)` or :math:`(minibatch, C, d_1, d_2, ..., d_K)`
-    with :math:`K \geq 1` for the `K`-dimensional case (described later).
+    with :math:`K \geq 1` for the `K`-dimensional case. The latter is useful for
+    higher dimension inputs, such as computing NLL loss per-pixel for 2D images.
```

Comment:
Great point, and I agree with you. AFAICT it doesn't make mathematical sense to define "soft labels" for `NLLLoss` since likelihood requires a single specific outcome, as you mentioned. I'll update the PR to be `CrossEntropyLoss`-only.

Note that this will make `CrossEntropyLoss` no longer equivalent to `LogSoftmax` + `NLLLoss`; the docs will need to reflect that they are only equivalent for the case of class index targets.